src/Microsoft.Identity.Web/TokenCacheProviders/MsalAbstractTokenCacheProvider.cs

```diff
@@ -13,6 +21,27 @@ namespace Microsoft.Identity.Web.TokenCacheProviders
     /// <seealso cref="Microsoft.Identity.Web.TokenCacheProviders.IMsalTokenCacheProvider" />
     public abstract class MsalAbstractTokenCacheProvider : IMsalTokenCacheProvider
     {
+        private readonly IDataProtector? _protector;
+        private const string DefaultPurpose = "msal_cache";
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="serviceProvider">Service provider. Can be null, in which case the token cache
```

Comment:
There is an option to turn on data encryption `bool Encrypt`. I think it might be cleaner to: 

- always pass in a service provider, which could be null in some token cache implementations.
- check the Encrypt flag. If it's set and a IDataProtectionProvider exists- use it. If it doesn't exist - throw a meaningful exception.

Reply:
fixed.